### PR TITLE
Yel2462 iam

### PR DESCRIPTION
**Pull request by JWanderema**

### CODE REVIEW

File: iam_cdlz.tf [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            "Sid": "UseKMSKey",
+            "Effect": "Allow",
+            "Action": [
+                "kms:Encrypt",
+                "kms:Decrypt",
+                "kms:ReEncrypt*",
+                "kms:GenerateDataKey*",
+                "kms:DescribeKey"
+            ],
+            "Resource": "arn:aws:kms:eu-west-2:337779336338:key/ae75113d-f4f6-49c6-a15e-e8493fda0453"
```
Comment (david-wirelab) [inline iam_cdlz.tf]: This is not the right KMS key.

Comment (JWanderema) [reply iam_cdlz.tf]: Removed hard coding and added variable settings for kms keys. Also expanded policy.

File: iam_cdlz.tf [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            "Action": [
+                "kms:Encrypt",
+                "kms:Decrypt",
+                "kms:ReEncrypt*",
+                "kms:GenerateDataKey*",
+                "kms:DescribeKey"
+            ],
+            "Resource": "${aws_kms_key.bucket_key.arn}"
+        }
+
```
Comment (david-wirelab) [inline iam_cdlz.tf]: There's an extra line and a comma is missing.

File: iam_cdlz.tf [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            "Sid": "PutS3Bucket",
+            "Effect": "Allow",
+            "Action": [
+                "s3:GetObject",
+                "s3:PutObject",
+                "s3:DeleteObject"
+            ],
+            "Resource": "${aws_s3_bucket.freight_archive_bucket.arn}/archive/*"
+        },
+        {
```
Comment (david-wirelab) [inline iam_cdlz.tf]: The KMS policy shows up twice - one of them should be removed.

File: iam_cdlz.tf (hunk 2 above)
Comment (JWanderema) [reply iam_cdlz.tf]: removed

File: iam_cdlz.tf (hunk 3 above)
Comment (JWanderema) [reply iam_cdlz.tf]: removed repetition

**Pull request merged**